presto-main/src/main/java/com/facebook/presto/execution/StateMachine.java

```diff
@@ -86,6 +86,8 @@ public StateMachine(String name, Executor executor, T initialState, Iterable<T>
         this.terminalStates = ImmutableSet.copyOf(requireNonNull(terminalStates, "terminalStates is null"));
     }
 
+    // state changes are atomic and state is volatile, so a direct read is safe here
+    @SuppressWarnings("FieldAccessNotGuarded")
```

Comment:
How do you run these checks? Are those done by Intellij?

Reply:
Yes they are part of IntelliJ